src/ucs/sys/ptrcache.h

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2018.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_PTR_CACHE_H_
+#define UCS_PTR_CACHE_H_
+
+#include <ucs/datastruct/pgtable.h>
+#include <ucs/datastruct/list.h>
+#include <ucs/stats/stats_fwd.h>
+#include <ucm/api/ucm.h>
+
+typedef struct ucs_ptrcache         ucs_ptrcache_t;
+typedef struct ucs_ptrcache_region  ucs_ptrcache_region_t;
+
+
+struct ucs_ptrcache_region {
+    ucs_pgt_region_t    super;    /**< Base class - page table region */
+    ucs_list_link_t     list;     /**< List element */
+    ucm_mem_type_t      mem_type; /**< Memory type the pointer belongs to */
+};
+
+
+struct ucs_ptrcache {
+    pthread_rwlock_t      lock;       /**< protests the page table */
+    ucs_pgtable_t         pgtable;    /**< Page table to hold the regions */
+    char                  *name;
```

Comment:
i think name is not needed in this case, this data structure is pretty unique, basically we even need just one per process because unlike regcache it's not related to a particular device/md